flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/VectorizedColumnBatch.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.dataformat.vector;
+
+import org.apache.flink.table.dataformat.Decimal;
+import org.apache.flink.table.dataformat.vector.BytesColumnVector.Bytes;
+
+import java.io.Serializable;
+
+/**
+ * A VectorizedColumnBatch is a set of rows, organized with each column as a vector. It is the
+ * unit of query execution, organized to minimize the cost per row.
+ *
+ * <p>{@code VectorizedColumnBatch}s are influenced by Apache Hive VectorizedRowBatch.
+ */
+public class VectorizedColumnBatch implements Serializable {
+	private static final long serialVersionUID = 8180323238728166155L;
+
+	/**
+	 * This number is carefully chosen to minimize overhead and typically allows
+	 * one VectorizedColumnBatch to fit in cache.
+	 */
+	public static final int DEFAULT_SIZE = 2048;
+
+	private int numRows;
+	public final ColumnVector[] columns;
+
+	public VectorizedColumnBatch(ColumnVector[] vectors) {
+		this.columns = vectors;
+	}
+
+	/**
+	 * Resets the batch for writing.
+	 */
+	public void reset() {
+		for (ColumnVector column : columns) {
+			column.reset();
+		}
+		this.numRows = 0;
+	}
+
+	public void setNumRows(int numRows) {
```

Comment:
why not set this in constructor?

Reply:
`VectorizedColumnBatch` is reused usually, need `setNumRows` after updating it.

Reply:
But the `columns` is immutable, how do you plan to reuse `VectorizedColumnBatch`?

Reply:
No, `columns` is not immutable, we can update the data (like int[]), and reuse it. (not use `ColumnarRow` to update it, we direct copy data to int[] to speed up updating)